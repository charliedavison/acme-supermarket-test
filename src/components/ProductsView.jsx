import React from 'react';
import PropTypes from 'prop-types';

import Product from './Product.jsx';

class ProductsView extends React.Component {
    constructor(props) {
        super(props);
    };

    render() {
        return (
            <div className="products-container">
                {Object.keys(this.props.products).map(product => {
                    return (
                        <Product
                            key={product}
                            product={this.props.products[product]}
                            onClick={this.props.onClickProduct} />
                    );
                })}
            </div>
        );
    };
};
ProductsView.defaultProps = {
    products: {}
};
ProductsView.propTypes = {
    products: PropTypes.object.isRequired,
    onClickProduct: PropTypes.func.isRequired
};
module.exports = ProductsView;

