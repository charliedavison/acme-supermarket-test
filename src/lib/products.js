module.exports = {
    'FR1': {
        productCode: 'FR1',
        name: 'Fruit tea',
        price: 3.11,
        image: 'tea.jpg'
    },
    'SR1': {
        productCode: 'SR1',
        name: 'Strawberries',
        price: 5.00,
        image: 'strawberries.jpg'
    },
    'CF1': {
        productCode: 'CF1',
        name: 'Coffee',
        price: 11.23,
        image: 'coffee.jpg'
    }
};