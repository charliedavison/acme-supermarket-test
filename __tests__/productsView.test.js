import React from 'react';
import { shallow } from 'enzyme';
import ProductsView from './../src/components/ProductsView.jsx';
import products from './../src/lib/products.js';

describe('<ProductsView />', () => {
    let component;
    
    let props = {
        products,
        onClickProduct: jest.fn()
    };

    beforeEach(() => {
        component = shallow(<ProductsView {...props} />);
    });

    afterEach(() => {
        component = null;
    });

    test('Should match snapshot', () => {
        expect(component).toMatchSnapshot();
    });

    test('Should contain correct number of products', () => {
        const numberOfProducts = Object.keys(products).length;
        expect(component.find('.products-container').children()).toHaveLength(numberOfProducts);
    });

});