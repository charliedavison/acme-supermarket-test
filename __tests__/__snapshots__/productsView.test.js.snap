// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductsView /> Should match snapshot 1`] = `
<div
  className="products-container"
>
  <Product
    key="FR1"
    onClick={[MockFunction]}
    product={
      Object {
        "image": "tea.jpg",
        "name": "Fruit tea",
        "price": 3.11,
        "productCode": "FR1",
      }
    }
  />
  <Product
    key="SR1"
    onClick={[MockFunction]}
    product={
      Object {
        "image": "strawberries.jpg",
        "name": "Strawberries",
        "price": 5,
        "productCode": "SR1",
      }
    }
  />
  <Product
    key="CF1"
    onClick={[MockFunction]}
    product={
      Object {
        "image": "coffee.jpg",
        "name": "Coffee",
        "price": 11.23,
        "productCode": "CF1",
      }
    }
  />
</div>
`;
